includes/admin/llms.functions.admin.php

```diff
@@ -288,3 +288,96 @@ function llms_merge_code_button( $target = 'content', $echo = true, $codes = arr
 	return $html;
 
 }
+
+/**
+ * Deletes all pending orders after held duration.
+ *
+ * @since [version]
+ *
+ * @return void
+ */
+function llms_delete_pending_orders() {
+
+	$days = get_option( 'lifterlms_pending_orders_deletion' );
+
+	if ( ! $days ) {
+		return;
+	}
+
+	// Fetching llms_orders with post_status `llms-pending` created more than $days ago.
+	$pending_orders = get_posts(
+		array(
+			'post_type'      => 'llms_order',
+			'post_status'    => 'llms-pending',
+			'posts_per_page' => -1, // to get all posts.
+			'date_query'     => array(
+				array(
+					'before' => $days . ' days ago',
+				),
+			),
+		)
+	);
+
+	// Delete the pending orders.
+	foreach ( $pending_orders as $order ) {
+		wp_delete_post( $order->ID, true );
+	}
+
+}
+add_action( 'llms_delete_pending_orders', 'llms_delete_pending_orders', 10 );
+
+/**
+ * Deletes all inactive accounts without any enrollments after held duration.
+ *
+ * @since [version]
+ *
+ * @return null
+ */
+function llms_delete_inactive_accounts() {
+
+	$days = get_option( 'lifterlms_inactive_accounts_deletion' );
+
+	if ( ! $days ) {
+		return;
+	}
+
+	// Get all users with role `student` created more than $days ago.
+	$users = get_users(
+		array(
+			'role'          => 'student',
+			'fields'        => array( 'ID' ),
+			'number'        => -1, // to get all users.
+			'date_query'    => array(
+				array(
+					'before' => $days . ' days ago',
+				),
+			),
+			'count_total'   => false,
+			'fields'        => 'all',
+			'no_found_rows' => true,
+		)
+	);
+
+	// Check if users has any enrollments (courses or memberships).
+	$inactive_users = array_filter(
+		$users,
+		function( $user ) {
+
+			$student     = llms_get_student( $user->ID );
+			$enrollments = $student->get_courses()['found'] || $student->get_memberships()['found'];
```

Comment:
As said, we're fine with just one course and one membership, plus I'd rather check on 'results', not on 'found', in that case...

Reply:
Done.

Reply:
What did you do?
you removed the check on the memberships, why?
It was:
`$enrollments = $student->get_courses()['found'] || $student->get_memberships()['found'];`
I said: **we're fine with just one course and one membership, plus I'd rather check on 'results', not on 'found', in that case...**
you turned it into:
`$enrollments = $student->get_courses()['results'];`
while I only asked for something like this: 
`$enrollments = $student->get_courses( array( 'limit' => 1 ) )['results'] || $student->get_memberships( array( 'limit' => 1 ) )['results'];`

so limit to only 1 course or only 1 membership, and check `results` rather than `found`.
Can you fix this?

Reply:
Oh, I didn't understand the first time and didn't know that we could pass the `limit` param also. Thanks, it's done. 👍